### DISCUSSION TITLE
Possible to add custom keys to crashlytics reports

### DISCUSSION THREAD

**Discussion by Nikooos** [invertase/react-native-firebase#5435] [Q&A] [answered]
In the native SDK it is possible to add custom keys to the Crashlytics reports by using setCustomKey or setCustomValue. I don't see this (yet) in the react native library as far as I know. Of course I could write a bridge and send it through the native sdk, but I'm not using a bridge now and it would bring extra complexity to my app. 

What I want to do is send the current redux state with the crash by setting the latest state as a custom value on every state change. This way I know which state the app is in when the crash occurs. For now I'm using the logs, but that's not the way I want to go.

Am I missing something or do you have other suggestions on how to handle this? 

https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=ios#add-keys

**Comment by mikehardy** [comment, marked as answer]
Hi there! Internally, setAttribute and setAttributes are mapped to setCustomKey(s) 

https://rnfirebase.io/reference/crashlytics#setAttribute
https://github.com/invertase/react-native-firebase/blob/46556de7ce17ef1d55b52222a907732353addde6/packages/crashlytics/lib/index.js#L89
https://github.com/invertase/react-native-firebase/blob/46556de7ce17ef1d55b52222a907732353addde6/packages/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/ReactNativeFirebaseCrashlyticsModule.java#L110-L132

The code's always available to check and it's organized pretty well here, knowing the same may be useful in the future in case you need it

Not sure why our API naming doesn't match up exactly yet - the project's been running for years so it's probably some historical artifact -  but I noted it myself a while ago. If you wanted to add to index.d.ts / index.js a set of APIs called setAttribute/setAttributes that had the same javascript implementation, then just call the current ones from the *correct* naming that'd be a great PR for future folks

**Reply by Nikooos** [reply]
I'm sorry, I stopped diving in the code when I saw the name. But that is exactly what I need. 

I'll try to make a nice PR for it. Thanks for the clarification.

**Reply by mikehardy** [reply]
Great! Glad that helped, and sorry again for the API mismatch, leads to obvious confusion. No obligation of course but if you did post that PR I would give you all of may available internet points for the day :laughing: :smile:
